src/net.cpp

```diff
@@ -1841,18 +1841,35 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         if (interruptNet)
             return;
 
-        // Add seed nodes if DNS seeds are all down (an infrastructure attack?).
-        // Note that we only do this if we started with an empty peers.dat,
-        // (in which case we will query DNS seeds immediately) *and* the DNS
-        // seeds have not returned any results.
-        if (addrman.size() == 0 && (GetTime() - nStart > 60)) {
-            static bool done = false;
-            if (!done) {
-                LogPrintf("Adding fixed seed nodes as DNS doesn't seem to be available.\n");
+        // Static bool makes sure we only add fixed seeds once in binary lifetime.
+        static bool fixedSeedsAdded = false;
+        if (!fixedSeedsAdded && addrman.size() == 0) {
+            // When the node starts with an empty peers.dat, there are a few other sources of peers before
+            // we fallback on to fixed seeds: -dnsseed, -seednode, -addnode
+            // If none of those are available, we fallback on to fixed seeds immediately, else we allow
+            // 60 seconds for any of those sources to populate addrman.
+            bool addFixedSeeds = false;
+            // It is cheapest to check if enough time has passed first.
+            if (GetTime<std::chrono::seconds>() > nStart + std::chrono::minutes{1}) {
+                addFixedSeeds = true;
+                LogPrintf("Adding fixed seeds as 60 seconds have passed and addrman is empty.\n");
+            }
+
+            const bool dnsseed = gArgs.GetBoolArg("-dnsseed", DEFAULT_DNSSEED);
+            // Checking !dnsseed is cheaper before locking 2 mutexes.
```

Comment:
`GetBoolArg` takes the `cs_args` lock, so probably better to move this check outside of the loop if you care about that

Reply:
Done